### ISSUE TITLE
Too many requests

### ISSUE THREAD

**Issue by lafiosca** [diessica/slack-delete-files#9]
When running the script to delete hundreds of files, I am encountering this error from the Slack API:

```
Error while deleting files. { HTTPError: Response code 429 (Too Many Requests)
    at stream.catch.then.e (/Users/xxx/.nvm/versions/node/v6.10.3/lib/node_modules/slack-delete-files/node_modules/got/index.js:123:13)
    at process._tickCallback (internal/process/next_tick.js:109:7)
  message: 'Response code 429 (Too Many Requests)',
  host: 'slack.com',
  hostname: 'slack.com',
  method: 'POST',
  path: '/api/files.delete',
  statusCode: 429,
  statusMessage: 'Too Many Requests' }
```

I wonder if the script needs to do some throttling.

**Comment by mattbailey** [comment]
I had this problem too, and couldn't fine the repo here at first, so I duped the project.  I inserted rate limiting into my version and env based config:

https://www.npmjs.com/package/slack-expire-files

**Comment by lafiosca** [comment]
I've been meaning to release my own repo containing an AWS CloudFormation stack for deploying this in an automated fashion as well. Your improvements look useful.

**Comment by diessica** [comment]
Fixed in [v2.0.0](https://github.com/diessica/slack-delete-files/releases/tag/v2.0.0).
